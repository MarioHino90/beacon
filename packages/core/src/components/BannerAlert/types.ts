import React from "react";
import { ThemedStyledProps } from "styled-components";

import { UntransitionedBannerAlertProps } from "../UntransitionedBannerAlert";

export type Intent = "success" | "warning" | "danger" | "info";

export interface BannerAlertProps
  extends UntransitionedBannerAlertProps,
    Omit<StyledBannerTransitionProps, "children"> {}

export interface StyledBannerTransitionProps
  extends React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}

export interface IntentContextProviderProps {
  intent: Intent;
}

// this uses <any, any> because the extra props in the themed styled component
// do not matter, just that it is themed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface IntentColorArgs extends ThemedStyledProps<any, any> {
  intent: Intent;
}
