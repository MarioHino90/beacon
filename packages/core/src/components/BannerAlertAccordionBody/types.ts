export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
