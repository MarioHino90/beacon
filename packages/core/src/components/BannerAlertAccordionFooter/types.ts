export interface BannerAlertAccordionFooterProps
  extends React.HTMLAttributes<HTMLDivElement> {}
