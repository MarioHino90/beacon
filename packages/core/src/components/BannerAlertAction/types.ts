import { ButtonProps } from "../Button";
import { Intent } from "../BannerAlert";

export interface BannerAlertActionProps
  extends Omit<ButtonProps, "variant" | "intent"> {
  /** * the visual intent */
  intent?: Intent;
}
