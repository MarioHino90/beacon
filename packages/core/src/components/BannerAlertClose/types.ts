import { Intent } from "../BannerAlert";

export interface BannerAlertCloseProps {
  /**
   * visual banner intents */
  intent: Intent;
  /**
   * on close callack */
  onClose: () => void;
}

export interface StyledProps {
  /**
   * visual banner intents */
  intent: Intent;
}
