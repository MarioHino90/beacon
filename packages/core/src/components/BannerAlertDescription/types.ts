export interface BannerAlertDescriptionProps
  extends React.HTMLAttributes<HTMLDivElement> {}
