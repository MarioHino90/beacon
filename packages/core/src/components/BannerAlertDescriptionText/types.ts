export interface BannerAlertDescriptionTextProps
  extends React.HTMLAttributes<HTMLSpanElement> {}
