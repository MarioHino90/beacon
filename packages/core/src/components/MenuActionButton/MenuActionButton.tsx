import React from "react";
import classNames from "classnames";
import { getColor } from "@braze/beacon-styling";

import { BodyText } from "../BodyText";
import { Flex } from "../Flex";
import { Icon } from "../Icon";
import { MenuButton } from "../MenuButton";

import { MenuActionButtonProps } from "./types";

export const MenuActionButton = ({
  children,
  className,
  iconName,
  ...props
}: MenuActionButtonProps): React.ReactElement => (
  <li role="menuitem">
    <MenuButton
      {...props}
      className={classNames(["bcl-menu-action-button", className])}
      aria-label={
        typeof children === "string" ? children.toLowerCase() : undefined
      }
    >
      {iconName && (
        <Flex style={{ minWidth: 32 }} justifyContent="center">
          <Icon name={iconName} color={getColor("gray", 600)} size="1x" />
        </Flex>
      )}
      <BodyText size="md">{children}</BodyText>
    </MenuButton>
  </li>
);

MenuActionButton.displayName = "MenuActionButton";
