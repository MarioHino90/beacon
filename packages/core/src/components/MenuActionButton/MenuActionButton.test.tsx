import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";

import { MenuActionButtonProps } from "./types";
import { MenuActionButton } from "./MenuActionButton";

faker.seed(1603821058436);

describe("MenuActionButton", () => {
  let props: MenuActionButtonProps;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<MenuActionButton {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot when 'iconName' is defined", () => {
    const wrapper = shallow(<MenuActionButton {...props} iconName="file" />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot when 'children' is NOT a string", () => {
    const wrapper = shallow(
      <MenuActionButton {...props}>
        <span>hi</span>
      </MenuActionButton>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
