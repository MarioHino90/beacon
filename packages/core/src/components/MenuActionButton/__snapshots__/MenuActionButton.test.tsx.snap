// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuActionButton should match the snapshot 1`] = `
<li
  role="menuitem"
>
  <MenuButton
    aria-label="administrator vatu"
    className="bcl-menu-action-button turn-key"
    id="defect"
  >
    <BodyText
      size="md"
    >
      Administrator Vatu
    </BodyText>
  </MenuButton>
</li>
`;

exports[`MenuActionButton should match the snapshot when 'children' is NOT a string 1`] = `
<li
  role="menuitem"
>
  <MenuButton
    className="bcl-menu-action-button Rupee"
    id="Card"
  >
    <BodyText
      size="md"
    >
      <span>
        hi
      </span>
    </BodyText>
  </MenuButton>
</li>
`;

exports[`MenuActionButton should match the snapshot when 'iconName' is defined 1`] = `
<li
  role="menuitem"
>
  <MenuButton
    aria-label="mozambique"
    className="bcl-menu-action-button invoice"
    id="analyzer"
  >
    <Flex
      justifyContent="center"
      style={
        {
          "minWidth": 32,
        }
      }
    >
      <Icon
        color={[Function]}
        name="file"
        size="1x"
      />
    </Flex>
    <BodyText
      size="md"
    >
      Mozambique
    </BodyText>
  </MenuButton>
</li>
`;
